core/CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ endif()
 
 cmake_policy(SET CMP0054 NEW)
 cmake_policy(SET CMP0057 NEW)
+cmake_policy(SET CMP0079 NEW)
```

Comment:
```suggestion
cmake_policy(SET CMP0054 NEW) # Only interpret if() arguments as variables or keywords when unquoted.
cmake_policy(SET CMP0057 NEW) # Support new if() IN_LIST operator.
cmake_policy(SET CMP0079 NEW) # target_link_libraries() allows use with targets in other directories.
```

Reply:
Also I think we should request  `cmake_minimum_required(VERSION 3.17)`

Reply:
we do that in the toplevel CMakeLists.txt

Reply:
I unified cmake_policy to require 3.17 and enable all policies up to and including 3.19.
We also require CMake 3.17 now and don't use `cmake_minimum_required()` everywhere anymore.